### PR TITLE
Resolved issue 237

### PR DESCRIPTION
**Pull request by pawannec**

### CODE REVIEW

**Review by amitrnec — APPROVED** [review]

**Review by RupamNEC — APPROVED** [review]

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/EntityRequest.java [hunk 1]
```diff
@@ -20,18 +20,17 @@
 	protected String entityWithoutSysAttrs;
 	protected String keyValue;
 	protected String operationValue;
-	
+	protected String updAttrName;
```
Comment (ScorpioBroker) [inline Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/EntityRequest.java]: remove and save the original payload / construct it on partial update
original payload in the given example on general update/append would be 
{"brandName": {
    "value": "BMW",
    "type": "Property"
}
}
on partial it's only
{
    "value": "BMW",
    "type": "Property"
}
but you can construct it in this case.

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/serialization/EntityRequestGsonAdapter.java [hunk 2]
```diff
@@ -55,6 +58,8 @@ public EntityRequest deserialize(JsonElement json, Type typeOfT, JsonDeserializa
 				result.setOperationValue(entry.getValue().getAsString());
 			} else if (entry.getKey().equals(AppConstants.REQUEST_HD)) {
 				result.setHeaders(deserializeHeaders(entry.getValue()));
+			} else if (entry.getKey().equals(AppConstants.REQUEST_UPD_ATTR_NAME)) {
```
Comment (ScorpioBroker) [inline Commons/src/main/java/eu/neclab/ngsildbroker/commons/serialization/EntityRequestGsonAdapter.java]: remove and forward original payload instead

File: Core/SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionService.java [hunk 3]
```diff
@@ -857,7 +867,7 @@ private void checkSubscriptionsWithUpdate(EntityRequest updateRequest, long mess
 			e.printStackTrace();
 		}
 		update.setType(type);
-
+		update.setAttrname(updateRequest.getUpdAttrName());
```
Comment (ScorpioBroker) [inline Core/SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionService.java]: remove and change line 859 to take the original payload and do the same for append

File: Core/SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionService.java [hunk 4]
```diff
@@ -665,22 +665,32 @@ private Entity generateNotificationEntity(Entity entity, SubscriptionRequest sub
...
 	}
 
-	private boolean shouldFire(Entity entity, SubscriptionRequest subscription) {
+     private boolean shouldFire(Entity entity, SubscriptionRequest subscription) {
+		
 		if (subscription.getSubscription().getAttributeNames() == null
 				|| subscription.getSubscription().getAttributeNames().isEmpty()) {
 			return true;
 		}
+
```
Comment (ScorpioBroker) [inline Core/SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionService.java]: remove entire block ... not needed when the proper original payload/delta entity is given here.

**Review by ScorpioBroker — CHANGES_REQUESTED** [review]
Hi,

this still doesn't work on normal append 
patch on
http://localhost:9090/ngsi-ld/v1/entities/urn:ngsi-ld:Vehicle:A100/attrs/
{"brandName": {
    "value": "123asdasd5asd67",
    "type": "Property"
}
}
does still trigger the notification where it shouldn't.

**Review by ScorpioBroker — APPROVED** [review]

**Pull request merged**